### DISCUSSION TITLE
How to get the a whole json object as string as the value instead of drilling down to properties

### DISCUSSION THREAD

**Discussion by Dunge** [grafana/grafana-infinity-datasource#587] [Q&A] [answered]
Hello, first of all thanks for this tool it is truly awesome.

My question will seems strange.
I'm trying to use this visualization plugin: https://grafana.com/grafana/plugins/orchestracities-map-panel/

But instead of using a PostGIS database as in their example, I would like to grab the geojson document from an HTTP GET query. Infinity datasource seems to be the best plugin I found to do that.

As far as I understand, the map panel plugin only requires a single entry that would be named "geojson" and as the value contains the whole document.

With infinity, the JSON parser kinda force you to have an object or array and actually parse it to returns a tabular dataset. Is there a way so I could trick it not to do any parsing and returns the whole document as is (or even better with a single entry called geojson)?

I thought the format "as is" dropdown could help but no.
I also tried to add a single column called geojson and use * as the selector but no.

**Comment by Dunge** [comment]
Actually the geojson format is a "FeatureCollection" containing an array of features. It would be better if each could be a separate entry, which I can do using "features" as the root. But then I need the parsing to stops there, and return the whole object as a json string, and not for example just the name as in most samples.

**Comment by Dunge** [comment]
This is the default:
![image](https://user-images.githubusercontent.com/6548184/234437453-a105a6e6-a30d-4d0a-8b95-e30e7a0778ce.png)

This is my try using a JSONPath selecting everything and adding a column, unfortunately it returns the whole array instead of separating them
![image](https://user-images.githubusercontent.com/6548184/234437741-14217d18-8201-4aa3-9934-5f43f2683ec7.png)

**Comment by Dunge** [comment]
Ok, I finally managed to get what I wanted by randomly trying to check the "is data in columnar format" checkbox:

![image](https://user-images.githubusercontent.com/6548184/234439251-bfbcf19d-d932-464a-ae08-7ed462755eb7.png)

Unfortunately, the orchestracities plugin doesn't seem to process it... might have to check with them.

**Reply by Dunge** [reply]
Reading about columnar format, I don't think this is supposed to be one. It's just a luck that it seems to work, but that's not a proper solution.

**Comment by Dunge** [comment, marked as answer]
Better solution based on [the screenshot here](https://community.grafana.com/t/adding-custom-geojson-to-geomap/57991/12) using UQL:

```
parse-json
| scope "features"
|project "geometry"
```

**Reply by yesoreyeram** [reply]
Glad you figured it out yourself
